app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.controllers
+
+import cats.data.EitherT
+import cats.implicits._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents, Result}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.tradergoodsprofiles.controllers.actions.{AuthAction, ValidateHeaderAction}
+import uk.gov.hmrc.tradergoodsprofiles.models.{APICreateRecordRequest, CreateRecordResponse, RouterCreateRecordRequest}
+import uk.gov.hmrc.tradergoodsprofiles.services.{DateTimeService, RouterService}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CreateRecordController @Inject() (
+  authAction: AuthAction,
+  validateHeaderAction: ValidateHeaderAction,
+  dateTimeService: DateTimeService,
```

Comment:
think we need to wait until 673 is merged in as we have removed datetimeservice now from error responses and refactored some of the code around it. 673 is nearly done is some test cases needs to be sorted, we can pair on this ticket and update the rest of changes.